src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkRuleOrCheckpoint.kt

```diff
@@ -5,4 +5,17 @@ import com.jetbrains.python.psi.PyTypedElement
 
 interface SmkRuleOrCheckpoint : SmkRuleLike<SmkRuleOrCheckpointArgsSection>, PyTypedElement {
     fun collectWildcards(): List<Pair<PsiElement, String>>
+
+    /**
+     * Snakemake wildcards can be defined by 3 sections.
+     * Defining section priority is: output > log > benchmark
+     * @return Section that defines wildcards or null
+     *         if there are no such sections
+     */
+    fun getWildcardDefiningSection() =
+        statementList.statements.filter {
+            it.name in SmkRuleOrCheckpointArgsSection.KEYWORDS_GENERATING_WILDCARDS
```

Comment:
If `it` is SmkRuleOrCheckpointArgsSection, please use `SmkRuleOrCheckpointArgsSection.sectionKeyword`. We have not guarantees, that `it.name` is sectionKeyword